dxr/plugins/clang/dxr-index.cpp

```diff
@@ -80,45 +79,57 @@ struct FileInfo {
   std::string realname;
   std::ostringstream info;
   bool interesting;
+  static std::string srcdir;  // the project source directory
```

Comment:
> Why persist the strange decision of making these singletons? Conceptually, are they not one-per-DXRIndexAction-instantiation?

Are you thinking that more than one DXRIndexAction could exist at the same time, in a single `clang++` call?  And that the different `DXRIndexActions` might have different `srcdir`, `outdir`, or `tmpdir` values?  If so then I agree that those values can't be static.  But I didn't think either of those things did/could happen, in which case I think it makes sense for at least `FileInfo::srcdir` and `FileInfo::outdir` to be static, since every one of the many `FileInfo`s created will be using the same value for `srcdir` and `outdir`.

It's never been entirely clear to me, but the test runs I've looked at seem to also indicate that only one `IndexConsumer` gets created per `DXRIndexAction`, so, if that's the case, then there's no need to make `IndexConsumer::tmpdir` static, except for the fact that if it isn't static, then `DXRIndexAction` will have to store its value from when it's created in `ParseArgs` to when it can be passed at `IndexConsumer` creation time in `CreateASTConsumer`.  But in this case (even if only one `IndexConsumer` is ever created) I still think it makes logical sense for `IndexConsumer::tmpdir` to be static, since every `indexConsumer` instance created by a given `DXRIndexAction` is going to use the same value (unless of course that value can change over the course of a given compile!).